ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml

```diff
@@ -66,26 +66,46 @@ tests:
     cluster_profile: azure-qe
     env:
       BASE_DOMAIN: qe.azure.devcluster.openshift.com
-      E2E_RUN_INTERMEDIATE_TAGS: '@4.11 and @azure-ipi and @upgrade-sanity and not
-        @fips'
       E2E_RUN_LATEST_TAGS: '@4.12 and @azure-ipi and @upgrade-sanity and not @fips'
       UPGRADE_CHECK_RUN_TAGS: '@4.10 and @azure-ipi and @upgrade-check and not @fips'
       UPGRADE_PRE_RUN_TAGS: '@4.10 and @azure-ipi and @upgrade-prepare and not @fips'
     test:
-    - chain: openshift-upgrade-qe-test-paused
+    - as: set-upgrade-releases
+      commands: echo "${RELEASE_IMAGE_INTERMEDIATE},${RELEASE_IMAGE_TARGET}" > ${SHARED_DIR}/upgrade-edge
+      dependencies:
+      - env: RELEASE_IMAGE_INTERMEDIATE
+        name: release:intermediate
+      - env: RELEASE_IMAGE_TARGET
+        name: release:target
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
```

Comment:
Perhaps we can move above newly added lines into a new step, so that it can be reused and integrated into `openshift-upgrade-qe-test-eus` chain to make it simple ?

Reply:
It is a bit hard, because the upgrade path depends on ci job config, each hop upgrade's image only get known in ci yaml file, and different ci jobs is using different ENV vars and its value is pointed to different release image.